pydantic_extra_types/coordinate.py

```diff
@@ -0,0 +1,187 @@
+from typing import Any, Callable, ClassVar, Tuple, Type, Union
+
+from pydantic import GetCoreSchemaHandler, GetJsonSchemaHandler
+from pydantic._internal import _repr
+from pydantic.json_schema import JsonSchemaValue
+from pydantic_core import CoreSchema, PydanticCustomError, core_schema
+
+LongitudeLatitudeType = Union[str, int, float]
+
+
+class Latitude(float):
+    min: ClassVar[float] = -90.00
+    max: ClassVar[float] = 90.00
+
+    def __init__(self, value: LongitudeLatitudeType):
+        self.validate_lat(value)
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: Type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.float_schema(ge=cls.min, le=cls.max)
+
+    @classmethod
+    def validate_lat(cls, value: LongitudeLatitudeType) -> None:
+        """
+        Validate the latitude value.
+
+        Raises:
+            PydanticCustomError: If the latitude value is not within the valid range.
+        """
+        try:
+            if float(value) > cls.max or float(value) < cls.min:
+                raise PydanticCustomError('latitude_error', 'Latitude must be between -90 and 90')
+        except ValueError:
+            raise PydanticCustomError(
+                'latitude_error',
+                'value is not a valid Latitude: value must be a str, int or float',
+            )
+
+
+class Longitude(float):
+    min: ClassVar[float] = -180.00
+    max: ClassVar[float] = 180.00
+
+    def __init__(self, value: LongitudeLatitudeType):
+        self.validate_long(value)
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: Type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.float_schema(ge=cls.min, le=cls.max)
+
+    @classmethod
+    def validate_long(cls, value: LongitudeLatitudeType) -> None:
```

Comment:
Shouldn't we be more explicit as well ? `validate_longitude` ?

Reply:
I've renamed, WDYT?

Reply:
LGTM

Reply:
I think that's very good as well.